src/main/java/com/vishva/Payapp/exception/ControllerAdvisor.java

```diff
@@ -0,0 +1,47 @@
+package com.vishva.Payapp.exception;
+
+import com.vishva.Payapp.error.ApiError;
+import com.vishva.Payapp.exchange.CreditResponse;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.FieldError;
+import org.springframework.validation.ObjectError;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.context.request.WebRequest;
+import org.springframework.web.servlet.mvc.method.annotation.ResponseEntityExceptionHandler;
+import java.util.ArrayList;
+import java.util.List;
+
+
+@ControllerAdvice
+public class ControllerAdvisor extends ResponseEntityExceptionHandler {
+
+    @ExceptionHandler(IllegalArgumentException.class)
+    public ResponseEntity<CreditResponse> handleAccountNotFoundException(Exception ex) {
+        CreditResponse creditResponse = new CreditResponse();
+        creditResponse.setCreditStatus(ex.getMessage());
+        return ResponseEntity.badRequest().body(creditResponse);
+    }
+
+    @Override
+    protected ResponseEntity<Object> handleMethodArgumentNotValid(
+            MethodArgumentNotValidException ex,
+            HttpHeaders headers,
+            HttpStatus status,
+            WebRequest request) {
+
+        List<String> errors = new ArrayList<String>();
+        for (FieldError error : ex.getBindingResult().getFieldErrors()) {
+            errors.add(error.getField() + ": " + error.getDefaultMessage());
+        }
+        for (ObjectError error : ex.getBindingResult().getGlobalErrors()) {
+            errors.add(error.getObjectName() + ": " + error.getDefaultMessage());
+        }
+
+        ApiError apiError = new ApiError(HttpStatus.BAD_REQUEST, ex.getLocalizedMessage(), errors);
+        return ResponseEntity.badRequest().body(apiError.getErrors());
```

Comment:
Here, in this response you still not returning the CreditResponse rather List<String>.


Below mentioned code would have done the job.
 CreditResponse creditResponse = new CreditResponse();
 creditResponse.setCreditStatus(ex.getBindingResult().getFieldErrors().get(0).getDefaultMessage());
 return ResponseEntity.badRequest().body(creditResponse);